### DISCUSSION TITLE
Is `prerender = true` search-query-sensitive?

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#8964] [Q&A] [answered]
means, does `prerender = true` renders one version for all search query variations per path or one version per search query variation per path? I guess latter, so...
```
example.com/products?category=2
example.com/products?category=3
```
...create two different prerendered pages, right?

**Comment by geoffrich** [comment, marked as answer]
Rendering multiple versions for different search param values is not possible -- prerendering outputs a single HTML file per page, and cannot know the value of search params at build time.

[The docs](https://kit.svelte.dev/docs/page-options#prerender-when-not-to-prerender) warn against accessing url.searchParams for prerendered pages:

> Accessing [url.searchParams](https://kit.svelte.dev/docs/load#using-url-data-url) during prerendering is forbidden. If you need to use it, ensure you are only doing so in the browser (for example in onMount).
